cpp/src/c_api/graph_sg.cpp

```diff
@@ -513,26 +513,19 @@ extern "C" cugraph_error_code_t cugraph_sg_graph_create(
     weight_type = cugraph_data_type_id_t::FLOAT32;
   }
 
-  // FIXME:  The combination of edge_ids != nullptr, edge_type_ids == nullptr
-  //         logically should be valid, but the code will currently break if
-  //         that is that is specified
-  CAPI_EXPECTS(
-    (edge_type_ids == nullptr && edge_ids == nullptr) ||
-      (edge_type_ids != nullptr && edge_ids != nullptr),
-    CUGRAPH_INVALID_INPUT,
-    "Invalid input arguments: either none or both of edge ids and edge types must be provided.",
-    *error);
+  CAPI_EXPECTS((edge_ids == nullptr) || (p_edge_ids->type_ == edge_type),
+               CUGRAPH_INVALID_INPUT,
+               "Invalid input arguments: Edge id type must match edge (src/dst) type",
```

Comment:
Same here. edge ID type should match with edge_t, not vertex_t.

Reply:
I changed the message, but it's still confusing to the user why they may get an error if their input graph has int64 vertex ids but only int32 edge ids.  It's not that crazy for that to happen (i.e. vertex ids are a hash).  I'll consider adding something at the Python layer to make that more clear.

Reply:
I just added the warning at the Python layer.